src/Tests/AddressTest.php

```diff
@@ -0,0 +1,89 @@
+<?php
+declare(strict_types=1);
+
+namespace GingerPluginSdk\Tests;
+
+use GingerPluginSdk\Entities\Address;
+use GingerPluginSdk\Properties\Country;
+use GingerPluginSdk\Exceptions\OutOfEnumException;
+use GingerPluginSdk\Exceptions\OutOfPatternException;
+use GingerPluginSdk\Helpers\SingleFieldTrait;
+use PHPUnit\Framework\TestCase;
+use TypeError;
+
+class AddressTest extends TestCase
+{
+    private Address $address;
+
+    public function __construct(?string $name = null, array $data = [], $dataName = '')
+    {
+        $this->address = new Address(
+            address_type: "customer",
+            postal_code: "38714",
+            street: "Soborna",
+            city: "Poltava",
+            country: new Country(
+                "UA"
+            )
+        );
+        self::assertSame((bool)$this->address, true);
+        parent::__construct($name, $data, $dataName);
+    }
+
+    public function test_address_type_invalid_enumeration_exception()
+    {
+        self::expectException(OutOfEnumException::class);
+        $this->address->getAddressType()->set('james_bond');
+    }
+
+    public function test_country_pattern_exception()
+    {
+        self::expectException(OutOfPatternException::class);
+        $this->address->getCountry()->set(new Country('NIGERIA'));
+    }
+
+    public function test_invalid_types_constructor_exception()
+    {
+        self::expectException(TypeError::class);
+        new Address(
+            address_type: "customer", postal_code: "38714", street: "Red", city: "Amsterdam", country: 4
+        );
+    }
+
+
+    public function test_set_housenumber()
+    {
+        $expected = $this->createSimpleField(
+            property_name: 'housenumber',
+            value: "30"
+        );
+
+        $this->address->setHousenumber("30");
+        self::assertSame($this->address->getHousenumber()->get(), $expected->get());
```

Comment:
Why does this not return a scalar value? I think there's actually no real value in a BaseField response for the consuming code. Typically for API options the only required method is something like to array.

Reply:
> Why does this not return a scalar value? I think there's actually no real value in a BaseField response for the consuming code. Typically for API options the only required method is something like to array.

Forgot to modify Address `getBlaBlaBla()` methods. I just thought also about you remark and some entities already updated (inside this version), but looks like not this :)

Reply:
You could check `Customer` entity. In Get methods i return scalar value